### ISSUE TITLE
send_keys sends wrong value

### ISSUE THREAD

**Issue by polaris007** [appium/appium#5331]
Issue https://github.com/appium/appium/issues/4844 is marked as closed. But I still encounter the same issue. It happens consistently in my env:

Test device: Samsung Galaxy S4, Android 4.4.2
Appium java client: 2.2.0
Appium server: 1.4.0 and 1.4.6

I sent "https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json", but always got "https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd.".

I invoke send_keys and then invoke getText to get the text of the element. The following are the logs of Appium:
----------------------------------------------------------------------------------------
info: --> POST /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/clear {"id":"2"}

info: [debug] Pushing command to appium work queue: ["element:clear",{"elementId":"2"}]

info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:clear","params":{"elementId":"2"}}

info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: clear

info: [debug] [BOOTSTRAP] [debug] Attempting to clear using UiObject.clearText().

info: --> GET /wd/hub/status {}

info: [debug] Responding to client with success: {"status":0,"value":{"build":{"version":"1.4.1","revision":"ce6220829063679bcccf5724515c29569c4a16d9"}},"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: <-- GET /wd/hub/status 200 2.046 ms - 155 {"status":0,"value":{"build":{"version":"1.4.1","revision":"ce6220829063679bcccf5724515c29569c4a16d9"}},"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: [debug] [BOOTSTRAP] [debug] Clearing text not successful. Attempting to clear by selecting all and deleting.

info: [debug] [BOOTSTRAP] [debug] Text remains after clearing, but it appears to be hint text.

info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}
info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: <-- POST /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/clear 200 9344.674 ms - 76 {"status":0,"value":true,"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: --> POST /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/value {"id":"2","value":["https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json"]}

info: [debug] Pushing command to appium work queue: ["element:setText",{"elementId":"2","text":"https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json","replace":false}]

info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:setText","params":{"elementId":"2","text":"https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json","replace":false}}

info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: setText
info: [debug] [BOOTSTRAP] [debug] Using element passed in.

info: [debug] [BOOTSTRAP] [debug] Attempting to clear using UiObject.clearText().

info: [debug] [BOOTSTRAP] [debug] Text remains after clearing, but it appears to be hint text.

info: [debug] [BOOTSTRAP] [debug] Text not cleared. Assuming remainder is hint text.

info: [debug] [BOOTSTRAP] [debug] Sending plain text to element: https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json

info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":true,"status":0}

info: [debug] Responding to client with success: {"status":0,"value":true,"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: <-- POST /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/value 200 7118.701 ms - 76 {"status":0,"value":true,"sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: --> GET /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/text {}

info: [debug] Pushing command to appium work queue: ["element:getText",{"elementId":"2"}]

info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getText","params":{"elementId":"2"}}
info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: getText

info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"https:\/\/bej301133.cn.oracle.com:W00a\/bmax\/bmconfig_oam_pwd.","status":0}

info: [debug] Responding to client with success: {"status":0,"value":"https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd.","sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: <-- GET /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/text 200 318.474 ms - 133 {"status":0,"value":"https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd.","sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

info: --> GET /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/text {}

info: [debug] Pushing command to appium work queue: ["element:getText",{"elementId":"2"}]

info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"element:getText","params":{"elementId":"2"}}

info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
info: [debug] [BOOTSTRAP] [debug] Got command action: getText

info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"https:\/\/bej301133.cn.oracle.com:W00a\/bmax\/bmconfig_oam_pwd.","status":0}

info: [debug] Responding to client with success: {"status":0,"value":"https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd.","sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}
info: <-- GET /wd/hub/session/c0eb53e0-68af-4691-b75a-d8fb14aae3fb/element/2/text 200 37.732 ms - 133 {"status":0,"value":"https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd.","sessionId":"c0eb53e0-68af-4691-b75a-d8fb14aae3fb"}

**Comment by Jonahss** [comment]
See if you can check to make sure the element exists before tapping on it. It's possible that it's disabled or not visible.

Try getting the location of the button, then creating a TouchAction to touch the screen with those coordinates.

**Comment by Jonahss** [comment]
oops, posted answer to the wrong ticket >.<

**Comment by Jonahss** [comment]
As @jlipps mentioned, it seems to be a bug in uiautomator.
This is what you can do: send the text. Then send a command to check the contents of the text input. If the contents don't equal the value you sent, clear the textbox and send your command again.

If this is affecting a lot of users we could build this check into appium, but that would slow down all tests :(

**Comment by polaris007** [comment]
@Jonahss and @jlipps ,
Checking and content and resending text doesn't work for me. I tried 20 times and got the wrong content every time. It happens consistently

**Comment by Jonahss** [comment]
Whoah!
Is it wrong in the exact same way each time? Or are the letters mixed up in different ways.

**Comment by polaris007** [comment]
Hi Jonahss,
It is always wrong in the exact same way. I sent "https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json", but always got "https://bej301133.cn.oracle.com:W00a/bmax/bmconfig_oam_pwd." on device. "9002" is always changed to "W00a" and "json" is lost

**Comment by Jonahss** [comment]
Ooh!
Ok I will try to reproduce. Didn't realize your issue was so predictable. there should be a way to fix this I hope.

**Comment by Jonahss** [comment]
Ok, I used the android [ApiDemos](https://github.com/appium/android-apidemos) app and used this command (javascript client, wd.js):

`element.sendKeys("https://bej301133.cn.oracle.com:9002/bmax/bmconfig_oam_pwd.json", console.log)`

It correctly typed the string exactly and didn't have any errors.

What language/locale setting is your device on? I did this on a Galaxy S4 with Android 4.4.2
Appium 1.4.7

If your local is set to English, maybe the issue is the editText field you are using? Is it the default android.widget.editText?

**Comment by polaris007** [comment]
Hi Jonah,
Appreciate for your efforts on this issue. 
Language on my device is English and the element is android.widget.EditText.
However, I found this issue has something to do with the input method. My previous default input method is a 3rd-party input method. After I switched to the built-in input method and updated some settings (e.g. disable AutoCapitalisation setting), the text can be input correctly now.
